src/compiler/config/validate-service-worker.ts

```diff
@@ -18,68 +18,70 @@ import type * as d from '../../declarations';
  * If we do find a service worker configuration we do some validation to ensure
  * that things are set up correctly.
  *
- * @param config the current, validated configuration
- * @param outputTarget the `www` outputTarget whose service worker
- * configuration we want to validate. **Note**: the `.serviceWorker` object
- * _will be mutated_ if it is present.
+ * @param config the current, validated Stencil config.
+ * @param serviceWorker The existing service worker property on the un-validated output target.
+ * @param appDir The validated `appDir` string for the output target.
+ * @returns `false` if explicitly set by the user, `null` if explicitly set by user or in development mode,
+ * or a {@link d.ServiceWorkerConfig} otherwise.
  */
-export const validateServiceWorker = (config: d.ValidatedConfig, outputTarget: d.OutputTargetWww): void => {
-  if (outputTarget.serviceWorker === false) {
-    return;
+export const validateServiceWorker = (
+  config: d.ValidatedConfig,
+  serviceWorker: d.OutputTargetWww['serviceWorker'],
+  appDir: string,
+): d.ValidatedOutputTargetWww['serviceWorker'] => {
```

Comment:
is `d.ValidatedOutputTargetWww['serviceWorker']` a different type than `d.OutputTargetWww['serviceWorker']` ? afaik they'll both resolve to `ServiceWorkerConfig | null | false`

also after thinking on it a bit I'm wondering if we want to have types like `ValidatedOutputTargetWww` hanging around (i.e. in `src/declarations/stencil-public-compiler.ts`) since I think if we did this for all of the output targets and other things we validate it could be quite a proliferation of types.

One alternative might be to inline `Required<OutputTargetWww>` in `src/compiler/config/outputs/validate-www.ts`. Just thinking that even if `validateWww` returns an array holding a union that includes `ValidatedOutputTargetWww` the `outputTargets` prop on `ValidatedConfig` is still going to be just `OutputTarget[]` so although we have some greater type assurance _while validating_ from `ValidatedOutputTargetWww` we would essentially type-hole that when `.outputTargets` is accessed later on...

I guess the "real" way to address that would be to introduce 'validated' versions of _all_ the output targets and then change the type of `ValidatedConfig['outputTargets']` to be `ValidatedOutputTarget[]` which is a big lift!

Anywhoo - just thinking that through a bit, lmk what you think

Reply:
I'm not tied to the validated target types by any means. I think they might make more sense down the road as we fix up types in all our validation logic since there _may_ be a case where the types between a validated and non-validated output target config are different other than being required... but idk. 

I know we **didn't** create a dedicated type for `dist` after fixing the SNCs in there, so happy to switch this back!